repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java

```diff
@@ -1243,4 +1245,54 @@ public void createAndQueueTask(String taskType, AtlasEdge relationshipEdge, Atla
 
         RequestContext.get().queueTask(task);
     }
+
+
+    public void resetHasLineage(Collection<AtlasEdge> edgesToBeDeleted) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("resetHasLineage");
+
+        for (AtlasEdge edgeToBeDeleted : edgesToBeDeleted) {
+            if (!isRelationshipEdge(edgeToBeDeleted)) {
```

Comment:
instead of checking !isRelationshipEdge(edgeToBeDeleted)
I suggest check edgeToBeDeleted's label in (PROCESS_OUTPUTS, PROCESS_INPUTS)